gateway-ha/src/main/java/io/trino/gateway/ha/router/TrinoQueryProperties.java

```diff
@@ -97,6 +97,7 @@ public class TrinoQueryProperties
     private Set<String> catalogSchemas = ImmutableSet.of();
     private boolean isNewQuerySubmission;
     private boolean isQueryParsingSuccessful;
+    private Optional<String> parseErrorMessage;
```

Comment:
Why do we need both isQueryParsingSuccessful and parseErrorMessage variables? Is it possible to remove isQueryParsingSuccessful and return the value by checking the size of parseErrorMessage on the fly?

Reply:
So if there isn't an errorMessage than it is considered successful? 
Hmm.. I prefer having a separate variable. @willmostly any thoughts?

Reply:
When does isQueryParsingSuccessful become true when parseErrorMessage is present?
When does isQueryParsingSuccessful become false when parseErrorMessage is empty? 

Having two variables is troublesome as it's easy to miss updating another one. Why do you prefer having a separate variable?

Reply:
Nowadays am used to using Golang as it always has error field to check before doing any further actions. 
- https://go.dev/blog/error-handling-and-go

Quite convenient should i say. 
But am ok with just having error message and removing boolean field.

Reply:
Any ideas on handling this? because when i do this
```java
    @JsonProperty("isQueryParsingSuccessful")
    public boolean isQueryParsingSuccessful()
    {
        return errorMessage.isEmpty();
    }
```
I get this error. As of now, isQueryParsingSuccessful is assigned but never accessed. 
There is no need to access the field in code, but i don't see any other better ways than simply adding @SuppressWarnings("unused")
```
gateway-ha/src/main/java/io/trino/gateway/ha/router/TrinoQueryProperties.java:99:21
java: [UnusedVariable] The field 'isQueryParsingSuccessful' is never read.
```